File: Boj.단계별/12_정렬/javascript/2751.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const nums = input.slice(1,input.length-1).reduce((acc,cur) => {acc.push(Number(cur)) ; return acc},[]);

const mergeSort = (arr) => {
  if (arr.length <= 1) return arr;
  const mid = Math.floor(arr.length/2);
  let left,right;
  left = mergeSort(arr.slice(0,mid));
  right = mergeSort(arr.slice(mid,arr.length));
  let i = j = k = 0;
  while ( i< left.length && j<right.length){
      if (left[i] <= right[j]) arr[k++] = left[i++];
      else arr[k++] = right[j++];
  }
  while ( i< left.length) arr[k++] = left[i++];
  while ( j< right.length) arr[k++] = right[j++];
  return arr;
}


mergeSort(nums);
result = nums.reduce((prev,cur) => {
    prev += (cur+"\n");
    return prev;
},"");
console.log(result)

File: Boj.단계별/12_정렬/python/2751.py
import sys
input = sys.stdin.readline
n = int(input())
nums = []
for _ in range(n):
    nums.append(int(input()))

def merge_sort(arr):
    if len(arr) <= 1: return arr
    mid = len(arr)//2
    left = merge_sort(arr[:mid])
    right = merge_sort(arr[mid:])
    i = j = k = 0
    while (i<len(left) and j<len(right)):
        if left[i] <= right[j]: 
            arr[k] = left[i]
            i += 1
        else: 
            arr[k] = right[j]
            j += 1
        k+=1
    while i<len(left):
        arr[k] = left[i]
        k+=1
        i+=1
    while j<len(right):
        arr[k] = right[j]
        k += 1
        j += 1
    return arr
merge_sort(nums)
for num in nums:
    print(num)